src/tests/p11test/p11test_case_common.c

```diff
@@ -144,6 +144,38 @@ add_supported_mechs(test_cert_t *o)
 			o->mechs[0].result_flags = 0;
 			o->mechs[0].usage_flags = CKF_SIGN | CKF_VERIFY;
 		}
+	} else if (o->type == EVP_PKEY_ED25519) {
+		if (token.num_ed_mechs > 0 ) {
+			o->num_mechs = token.num_ed_mechs;
+			for (i = 0; i <= token.num_ed_mechs; i++) {
+				o->mechs[i].mech = token.ed_mechs[i].mech;
```

Comment:
shouldn't this be `i < token.num_ed_mechs`? Same question below/above to the other for loops...

Reply:
yes. It should be.